README.md

```diff
@@ -607,7 +623,19 @@ _Examples_:
 
 Binary files can be specified in the configuration map in their base64 encoded format with a basename having a `.base64` extension. Such files are then converted back to binary as they are copied to their target path.
 
-When an /etc/sssd/sssd.conf file is included in the configuration map, the httpd pod automatically enables the sssd service upon startup.
+### Auth-type and auth-configuration specification matrix in a configmap:
+
+* auth-type depicts the Identity Provider external authentication is being configured against.
+* auth-configuration identifies which Httpd external authentication configuration files to load.
+
+| auth-type | auth-configuration | Note |
+|-----------|--------------------|------|
+| internal  | internal           | Database / ManageIQ Ldap(s) / Amazon    |
+| ldap      | external           |     |
```

Comment:
What happens if someone configures a nonsense combination of these parameters (e.g. ldap + active-directory or saml + external)? It seems like the only one with multiple combinations is `active-directory` so can we assume the `auth-configuration` value for the others?

Or even better, can we make `auth-configuration` specific to `active-directory`? Maybe something like `active-directory-config-set` or do we intend to add more `auth-configuration` options for other `auth-type`s in the future?

Reply:
The auth-configuration is directly drives what Apache configurations we support.  they cannot specify anything else. As we add support for others in the future, i.e. openid, we'd be adding Apache config/include files to reflect the new auth-configuration. So the auth-configuration set is strict/concrete.  The auth-type identifies the identify provider we're targeting, note how multiple types share the same auth-configuration apache include files.  As far as the values there, if we want/need to to differentiate via the auth-config CLI a generated config map for Keycloak vs. let's say Active Directory Federated Services, we can have auth-type (saml | adfs | okta | etc.), but the auth-configuration must be saml.

Hopefully this help a bit.

Reply:
Okay, that's fine. So we can't make auth-configuration more specific to AD (even though right now that is the only auth-type that could possibly have more than one valid auth-configuration). But my first question still stands. Are we doing anything to validate against nonsense combinations of auth-type + auth-configuration?

Reply:
After discussion, it seems like the values in the left column do not have any effect on the runtime of the application.

The values in the right column can be used for the same purpose as the auth-type is currently being used for (deciding whether to copy config files listed in `auth-configuration.conf`) as well as which specific httpd auth configuration files we should include in the `authentication.conf` file, so we can remove the values on the left side (but can keep using the existing auth-type name for the field).